recipes/hypergbm/meta.yaml

```diff
@@ -0,0 +1,71 @@
+{% set name = "HyperGBM" %}
+{% set version = "0.2.3.2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/DataCanvasIO/{{ name }}/archive/refs/tags/{{ version }}.tar.gz
+  sha256: a2a6509ecb86c38279183cf747d1e14aa8012319dd91788b81d17ce876bfa19a
+
+build:
+  number: 0
+  skip: True  # [py<36 or py>38]
```

Comment:
Why can't this be noarch?

Reply:
@dopplershift Thanks for your review!

Because the higher version *graphviz* will cause build failed if it is required with *pyarrow*(required by hypernets) at the same time on windows, so I limit *graphviz* to version *2.38* on windows with:

>    - graphviz =2.38  # [win]

and the conda-forge-linter say:

>     * `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.

So I remove  it.

Reply:
In addition, we only test hypergbm on python 3.6/3.7/3.8, so we limit the python version to 3.6~3.8.

Reply:
> Because the higher version graphviz will cause build failed if it is required with pyarrow(required by hypernets) at the same time on windows, so I limit graphviz to version 2.38 on windows with:

I don't understand. It doesn't look like `hyperGBM` uses graphviz at all? So I'm not sure why you're depending on graphviz?
If you really do need to depend on `graphviz` here, `pyarrow` doesn't use `graphviz` so I think there may be some different root cause problem for the build failure. Any chance you could put it back to noarch so we can see what the failure is exactly?

> In addition, we only test hypergbm on python 3.6/3.7/3.8, so we limit the python version to 3.6~3.8.

You can limit the allowed Python versions for `host` and `run` like you do any dependency when the package is noarch.

Reply:
> I don't understand. It doesn't look like hyperGBM uses graphviz at all? So I'm not sure why you're depending on graphviz?

HyperGBM depends on catboost, graphviz and plotly  are required by catboost [link](https://github.com/catboost/catboost/blob/2bdc1140c8402a3457d80333c9c3095be034487c/catboost/python-package/setup.py#L297). Because catboost does not list graphviz and plotly in their recipe, so I list them to avoid pip check broken.

Reply:
Review the updated recipe please, thanks!